.github/workflows/snyk-scan.yaml

```diff
@@ -0,0 +1,73 @@
+name: Snyk Security Vulnerability Scan
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+    branches:
+      - main
+
+jobs:
+  snyk_scan_test:
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - uses: snyk/actions/setup@master
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      
+      - name: Check changed Deps files
+        uses: tj-actions/changed-files@v35
+        id: changed-files
+        with:
+          files: | # This will match all the files with below patterns
+            pyproject.toml
+    
+      - name: Scan python dependencies
+        if: contains(steps.changed-files.outputs.all_changed_and_modified_files, 'pyproject.toml')
+        env:
+          SNYK_TOKEN: '${{ secrets.SNYK_TOKEN }}'
+        run: |
+          python3.10 -m pip install .
+          python3.10 -m pip freeze > requirements.txt
+          snyk test \
+            -d \
+            --file=requirements.txt \
+            --command=python3.10 \
+            --skip-unresolved \
+     
+  snyk_scan_monitor:
+    if: ${{ github.event_name == 'push' || github.event_name == 'workflow_dispatch' }}
```

Comment:
Please use two workflow files instead. This job (`snyk_scan_monitor`) can be one workflow triggered `on:` push and  WF dispatch requests. 

The `snyk_scan_test` can be triggered by pull requests event with the path filter. We can then avoid use of `tj-actions/changed-files`.

WDYT?

Reply:
Sure we can do that